src/plugins/query_enhancements/server/routes/index.ts

```diff
@@ -85,8 +85,14 @@ export function defineSearchStrategyRouteProvider(logger: Logger, router: IRoute
           const queryRes: IDataFrameResponse = await searchStrategy.search(context, req as any, {});
           return res.ok({ body: { ...queryRes } });
         } catch (err) {
+          let error;
+          try {
+            error = JSON.parse(err.message);
+          } catch (e) {
+            error = err;
+          }
           return res.custom({
-            statusCode: err.name,
+            statusCode: error.status,
```

Comment:
doesnt this mean we will return a 400 error instead of 200 when the job fails? Shouldnt we be returning a 200 here?

Reply:
Since the search strategy doesn't throw an error, the request won't reach this catch. Instead, it'll just be returned by the `res.ok({ body: { ...queryRes } });`.